frontend/src/index.jade

```diff
@@ -215,5 +222,5 @@ html
 
     script(src="static/js/api.js")
     script(src="static/js/v_summary.js")
-    script(src="static/js/v_authorship.js")
+    script(src="build/v_authorship.js")
```

Comment:
Why this change?

Reply:
The build file generated by browersify is located in the ```build``` path.

Reply:
Apparently this change has caused the CI to failed. Please ensure that the CI passed the test before requesting for review.

Reply:
> Apparently this change has caused the CI to failed. Please ensure that the CI passed the test before requesting for review.

No, the CI was failing because of [this](https://github.com/reposense/RepoSense/pull/532#discussion_r255380348), which I have also mentioned [here](https://github.com/reposense/RepoSense/pull/532#issuecomment-464006057).

Reply:
>No, the CI was failing because of [this](https://github.com/reposense/RepoSense/pull/532#discussion_r255380348), which I have also mentioned [here](https://github.com/reposense/RepoSense/pull/532#issuecomment-464006057).

My bad, was looking through phone during my lunch time earlier.

Anyway, its bad to have the JS all over the place. Instead of modifying it while its still in the source directory, can you modify it after its build? Or move it to the build directory directly?

E.g. `browserify -t vueify -e frontend/src/static/js/v_authorship.js -o frontend/build/static/js/v_authorship.js`
or  `browserify -t vueify -e frontend/build/static/js/v_authorship.js -o frontend/build/static/js/v_authorship.js`
whichever works.

Reply:
> E.g. `browserify -t vueify -e frontend/src/static/js/v_authorship.js -o frontend/build/static/js/v_authorship.js`
> or `browserify -t vueify -e frontend/build/static/js/v_authorship.js -o frontend/build/static/js/v_authorship.js`
> whichever works.

I tried both of this, it doesn't work.

Reply:
>I tried both of this, it doesn't work, not sure why.

I think the `build` has to be executed after `spuild` for it to work. Can you try if changing the order on `package.json` helps?

```
    "spuild": "spuild frontend",
    "build": ...
```

Reply:
> I think the `build` has to be executed after `spuild` for it to work. Can you try if changing the order on `package.json` helps?
> 
> ```
>     "spuild": "spuild frontend",
>     "build": ...
> ```

Tried this, but it still doesn't work. Maybe the file needs to be modified while it is in the source directly before the build so that browersify can bundle up the 'require' dependencies during build?

Reply:
@apoorva17 

Can fill me in? From what I read, I believe you are trying to add `build` after `spuild`?

Reply:
@yong24s I'm using ```browserify``` to generate a modified v_authorship.js file which bundles up dependencies for using ```require``` in browser. I'm currently using the command ```browserify -t vueify -e frontend/src/static/js/v_authorship.js -o frontend/src/build/v_authorship.js```, which creates and puts the modified v_authorship.js file in ```src/build```. However, @eugenepeh would like to modify it after its build/move modified file directly to build directly using ```browserify -t vueify -e frontend/src/static/js/v_authorship.js -o frontend/build/static/js/v_authorship.js```. I tried doing this, but it is not working, which is why I tried adding ```build``` after ```spuild```. But it still doesn't seem to work.

Reply:
> 
> [yong24s#12](https://github.com/yong24s/RepoSense/pull/12)
> https://ci.appveyor.com/project/yong24s/reposense/builds/22623766#L42
> 
> ^ Hmm, it seems like the CI are already doing this automagically 🤔

Yeah, this seems to show that ```spuild``` is already being executed before ```build``` regardless of ordering in ```package.json```, so it shouldn't be the cause of the above problem.

Reply:
According to the [Travis logs in your builds](https://travis-ci.org/reposense/RepoSense/jobs/497467138#L716), the tasks order are like this `npmRunSpuild -> zipReport -> npmRunBuild`.

I made [changes to the `build.gradle`](https://github.com/yong24s/RepoSense/pull/12) to reorder it to `npmRunSpuild -> npmRunBuild -> zipReport`.

But I found out that `npmRunBuild` has a hidden gotcha, where it depends on `processResources`. This breaks the correct order we define in #552.

```
$ ./gradlew run --dry-run

:processResources SKIPPED <-- Moved to first
:downloadNode SKIPPED
:npmInstall SKIPPED
:npmRunSpuild SKIPPED
:npmRunBuild SKIPPED
:zipReport SKIPPED
:compileJava SKIPPED
:classes SKIPPED
:run SKIPPED
```

I have exhaustively tried to the correct its order but it creates a circular dependency. After having given some thoughts, I tried to change its name and it works. 🎉 

```
$ ./gradlew run --dry-run

:downloadNode SKIPPED
:npmInstall SKIPPED
:npmRunSpuild SKIPPED
:npmRunBrowserify SKIPPED
:zipReport SKIPPED
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:run SKIPPED
```

Not sure if this is what you needed, you can check my changes [here](https://github.com/yong24s/RepoSense/pull/12).

Reply:
@yong24s i've implemented the changes u suggested and it works locally, but it doesn't seem to work when i push it to netlify.

Reply:
@yong24s this worked thanks!